x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -621,6 +621,23 @@ function isValidTimeInterval(value: string | undefined): boolean {
   return parseTimeIntervalForJob(value) !== null;
 }
 
+// The earliest start time for the datafeed should be the max(latest_record_timestamp, latest_bucket.timestamp + bucket_span).
+export function getEarliestDatafeedStartTime(
+  latestRecordTimestamp: number | undefined,
+  latestBucketTimestamp: number | undefined,
+  bucketSpan?: Duration | null | undefined
+): number | undefined {
+  if (latestRecordTimestamp !== undefined && latestBucketTimestamp !== undefined) {
+    // if bucket span is availble (e.g. 15m) add it to the latest bucket timestamp in ms
+    const latestBucketStartTime = bucketSpan
```

Comment:
`latestBucketStartTime` might be a confusing name for future readers of this code, as bucket timestamps _are_ the bucket start time, so `latestBucketTimestamp` _is_ the start time of the latest bucket we have results for.  When you add on the bucket span you're getting the start time of the first bucket we _don't_ have results for.

It might be clearer to rename `latestBucketStartTime` in this code to `adjustedBucketStartTime`.

Reply:
Renamed here https://github.com/elastic/kibana/pull/71976/commits/dcce6b69503b4cb2e2252e705097a3657dc776dc